src/driver/communication_interface.cc

```diff
@@ -168,6 +170,10 @@ void CommunicationInterface::SetRobotData(
   robot_data_.robot_state = robot_state;
   robot_data_.robot_plan_next_conf = robot_plan_next_conf;
   robot_data_.has_robot_data = true;
+  if(is_sim_){
+    franka::RobotMode current_mode {robot_data_.robot_state.robot_mode};
+    robot_data_.robot_state.robot_mode = current_mode;
```

Comment:
what's the purpose of this block? it just reads it out and sets it again.

need to run hooks also.